src/solidity-frontend/solidity_convert.cpp

```diff
@@ -531,7 +621,30 @@ bool solidity_convertert::get_var_decl(
       return true;
   }
 
-  bool is_state_var = ast_node["stateVariable"] == true;
+  if (
+    t.get("#sol_type") == "CONTRACT" &&
+    t.identifier().as_string() == prefix + current_contractName)
+  {
+    // contract Base{
+    //   Base x;
```

Comment:
How is this implemented in Solidity?

Reply:
```solidity
contract Base
{
    Base x;
    uint public y;
    constructor()
    {
        y = 1;
    }

    function test() public   {
        x = Base(0x9cF86D8D08bC34248210474C4B019befb0fE70fA);
        assert(x.y() == 1); 
    }
}
```
- We first deploy this contract, gather the address 
- We initialize x with the contract address and deploy it again.
- Run function `test` and the assertion holds.